keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -91,6 +102,8 @@ def pretokenize(text, lowercase, strip_accents, split):
     # Preprocess, lowercase, strip and split input data.
     if text.shape.rank == 0:
         text = tf.expand_dims(text, 0)
+    if split_on_cjk and split:
```

Comment:
Wow interesting, it looks like we were already splitting on these. I think in that case we could actually avoid this `regex_replace` call and just keep two different split regexes.

`WHITESPACE_AND_PUNCTUATION_REGEX` and `WHITESPACE_PUNCTUATION_AND_CJK_REGEX`

```
if split:
    if split_on_cjk:
         split_pattern = WHITESPACE_PUNCTUATION_AND_CJK_REGEX
    else:
         split_pattern = WHITESPACE_AND_PUNCTUATION_REGEX
    text = tf_text.regex_split(...)
```

Would this work?

Reply:
Hmmm, yeah. But we'll have to create a third REGEX, namely, `PUNCTUATION_AND_CJK_REGEX` to pass to `keep_delim_regex_pattern`. Isn't it easier to just keep `tf.regex_replace()`, then? What do you think?

Reply:
Pushing changes for now, let me know if it's better to revert back to the original.

Reply:
This looks fine to me. Original is simpler, but we probably should care about efficiency here (and cutting an op should help that).

Reply:
Cool!